ksql-parser/src/main/java/io/confluent/ksql/schema/ksql/DefaultSqlValueCoercer.java

```diff
@@ -36,10 +42,19 @@
           .build();
 
   public <T> Optional<T> coerce(final Object value, final SqlType targetType) {
-    if (targetType.baseType() == SqlBaseType.ARRAY
-        || targetType.baseType() == SqlBaseType.MAP
-        || targetType.baseType() == SqlBaseType.STRUCT
-    ) {
+    return doCoerce(value, targetType);
+  }
+
+  private static <T> Optional<T> doCoerce(final Object value, final SqlType targetType) {
```

Comment:
Does the type parameter really add anything here? It's not used in the params or in the body of the method.

Reply:
I think you have me convinced :) I'll change that to use the explicit cast.